docs/backend.md

```diff
@@ -2,40 +2,285 @@
-├── routes.php # Register routes with their middlewares here
-├── settings.php # Slim settings and required settings for the dependencies
-└── validation.php # Register validation rules here
-```
-
-# Developing new route
+* [`src/`](../api/src) Source directory. Not public
+    * [`JMP/`](../api/src/JMP) Application directory
+        * [`Controllers/`](../api/src/JMP/Controllers) Controllers to handle requests and build the responses
+        * [`Middleware/`](../api/src/JMP/Middleware) Custom [Middlewares](https://www.slimframework.com/docs/v3/concepts/middleware.html)
+        * [``Models/``](../api/src/JMP/Models) Raw data objects with the functionality to convert to an array
+        * [``Services/``](../api/src/JMP/Services) Business logic and database communication
+        * [``Utils/``](../api/src/JMP/Utils) Useful classes
+
+## Slim specific classes
+* Private
+    * [``dependencies.php``](../api/src/dependencies.php) Initialize services, controllers, middlewares and add them to the slim container
+    * [``settings.php``](../api/src/settings.php) Slim settings and required settings for the dependencies
+    * [``routes.php``](../api/src/routes.php) Register routes with their middlewares here
+* Public
+    * [``index.php``](../api/public/index.php) Application entry point
+
+
+## Authentication
+
+### Usage
+We use [JWT](https://jwt.io/) for authentication. The token must be submitted in the Authorization header using the Bearer schema.  
+Example:  
+
+```http_request
+Authorization: Bearer <token>
+```
+
+Read more at the [api specification](api-v1.md#authentication).
+### Settings
+To adjust the settings, look here:
+ * In [settings.php](../api/src/settings.php)
+ * [.env](../api/.env) and [.env docs](dotenv.md)
+
+### Implementation
+The submitted token is decoded by the [jwt-middleware](https://github.com/tuupola/slim-jwt-auth).  
+The custom [authentication middleware](../api/src/JMP/Middleware/AuthenticationMiddleware.php) checks the permissions of the enquirer using the `sub` entry of the decoded token.
+If the enquirer has got the right permissions for the route, he can pass.
+If no token is supplied or if the supplied token is invalid, a 401 is returned.
+In case the enquirer just hasn't got the required permissions, a 403 is returned.
+### Login
+To receive a new token, the user have to call the [login route](api-v1.md#login).  
+If the username and the password are valid, a new token is generated with the following service class [Auth.php](../api/src/JMP/Services/Auth.php).
+The password is hashed using [`password_hash`](https://secure.php.net/manual/en/function.password-hash.php) with the `PASSWORD_DEFAULT` option before it is stored in the database.
+To verify a password at a login, [`password_verify`](https://secure.php.net/manual/en/function.password-verify.php) is used.
+### Registration
+Only administrators can register new users. To register a new user, the [register route](api-v1.md) has to be called.
+
```

Comment:
Which route is meant? The create user route?

I'd remove the registration section as the documentation of endpoints should not be in this file.

Reply:
It's also part of the authentication and a requirement of the project. So I think it's not wrong to link the registration of the api specification here.

It's the create user route, so I will update the link.